### DISCUSSION TITLE
How wide does a shoulder need to be to count as a shoulder?

### DISCUSSION THREAD

**Discussion by endolith** [streetcomplete/StreetComplete#4541] [Q&A] [answered]
I frequently see cases where there's a white line, then maybe a few inches of black, then a curb, and the distance between the white line and curb varies as I'm driving, so I'm not sure what to put for these unless it's very obvious.  

StreetComplete says "not necessarily wide enough for a car to stop completely off the traffic lane".  

https://wiki.openstreetmap.org/wiki/Key:shoulder says:

> Unless otherwise tagged, a shoulder should be assumed paved, **wide enough to be used as an emergency refuge for cars**, and wide enough for through passage by bicycles.

which would seem to imply that a car needs to fit in the shoulder to count?

**Comment by mnalis** [comment]
I _think_ the intention of that wording is that is acceptable that some (small) part of the car parked on the shoulder remain on the driving lane, as long as it **_does not obstruct cars driving on that lane_**. 

E.g.:
- if `90%` of the car fits on the shoulder, and `10%` on the driving lane, it is probably OK and it **_is_** a shoulder.
- However, if only `20%` of the car fits on the "shoulder", and `80%` must remain on the line used for driving (which would likely stop all traffic in that lane!), then it is **_not shoulder_** at all.

Finding the exact line between those two extremes is a matter of personal judgement - would **_you_** feel secure stopping **_your_** car at such "shoulder", while other cars buzz on the lane beside it? If so, it is probably a shoulder. If not, it is not.

**Reply by endolith** [reply]
OK, so it's car-centric?  A narrow shoulder could be useful for bike travel

**Reply by mnalis** [reply]
> OK, so it's car-centric? 

Yes, car-centric, see the [shoulder=*](https://wiki.openstreetmap.org/wiki/Key:shoulder) wiki that StreetComplete follows. It also mentions alternatives like [verge=*](https://wiki.openstreetmap.org/wiki/Key:verge), but those are not supported by StreetComplete. 

Also note that the full SC explanation text is `A shoulder is a clearly demarcated emergency stopping lane that allows vehicles to stop safely completely beside the traffic lane, though not always (fully) on paved surface.`

So shoulder might be for example `70%` marked on asphalt, followed by `30%` compacted surface, as in this picture on the left side:
https://en.wikipedia.org/wiki/Shoulder_(road)#/media/File:M4_Heading_west_at_J11_(start_of_the_N6_near_Kinnegad.jpg

---

> A narrow shoulder could be useful for bike travel

[Shoulders](https://en.wikipedia.org/wiki/Shoulder_(road)) are not (at least in vast majority of cases) used for moving traffic, but for stopping in emergency. If some narrower lane exists on that road, and is wide enough for bicycle traffic (and bicycles are allowed to ride on it!) then those are not shoulders but [bicycle lanes](https://wiki.openstreetmap.org/wiki/Tag:cycleway=lane) and StreetComplete have separate [quest](https://wiki.openstreetmap.org/wiki/StreetComplete/Quests) for that: `Is there a bicycle lane or path here? What type?`

If it is wide enough for bicycle, but it is **illegal** to drive a bicycle on it, than it should **not** be mapped as bicycle lane, but maybe as [verge=*](https://wiki.openstreetmap.org/wiki/Key:verge) with appropriate [width=*](https://wiki.openstreetmap.org/wiki/Key:width) and [surface=*](https://wiki.openstreetmap.org/wiki/Key:surface) for example. But one would have to use some other, more complex [editor](https://wiki.openstreetmap.org/wiki/Editors) than StreetComplete to do that (for example [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci) if using mobile device)

**Reply by westnordost** [reply]
Not entirely, @mnalis . Basically the wiki page says everything that can be said about this tag. Yes, it is fuzzy. Yes, there is no hard limit (defined) how slim it may be to still count as a shoulder.

Refer to the talk page also, I did a little research how the `shoulder=*` tag is used in reality:

https://wiki.openstreetmap.org/wiki/Talk:Key:shoulder#Documenting_definition_of_shoulder:width

**Reply by mnalis** [reply]
Well, wiki does say at some point (in its current revision at least):
> Unless otherwise tagged, a shoulder should be assumed paved, wide enough to be used as an emergency refuge for cars, and wide enough for through passage by bicycles. 

But yeah, you did some useful research there :heart:. Looking deeper into Talk page, however, I'm even more confused about what is assumed by this tag, i.e. what useful purpose can it serve to data consumers in its current state (when different mappers mean wildly different things when they use the same tag). Especially if [shoulder:surface](https://taginfo.openstreetmap.org/keys/shoulder:surface)  **_and_** [shoulder:width](https://taginfo.openstreetmap.org/keys/shoulder:width) are _not_ mapped :confused:

**Reply by westnordost** [reply]
Yeah, it's really not in a good shape. That the definition is so blurry only came to the surface after i implemented that quest though. 

It would be better if shoulder was to be replaced or to be joined by other tags that allow the different types of "shoulders" to be defined more precisely.

There is the hard shoulder (paved), the soft shoulder (unpaved, but maintained) and the verge (grass, not maintained as emergency stopping). Ideally, they should each have their own tag. Currently,  at least the first two are muddled together.

**Comment by westnordost** [comment]
Did someone just delete a spam comment here or is this done automatically by GitHub (once an account is determined to be a spam account)?

**Reply by matkoniecz** [reply]
I deleted one. It was a blatant porn spam scam by now deleted https://github.com/LauraRobinson2189 account, I also reported account to GitHub. Earlier I got multiple phantom notifications across multiple discussions, across multiple repositories - I assume that it was also now deleted spam.

**Reply by mnalis** [reply]
I've received notifications for at least 8 other GitHub spams in StreetComplete repository (each by different account), but they were all gone by the time I got to them.

**Comment by matkoniecz** [comment, marked as answer]
shoulder quest was disabled in https://github.com/streetcomplete/StreetComplete/commit/04a976deee7679f62184ac1216359ac00894a674 - see https://community.openstreetmap.org/t/shoulder-tag-is-confusing/5185 and https://github.com/streetcomplete/StreetComplete/issues/4617#issuecomment-1310223162
